libs/protocol-client/src/lib/client/contribution.ts

```diff
@@ -82,6 +85,74 @@ export class Contribution extends BaseClient {
     return contributions.updateContribution;
   }
 
+  public async bulkMint(
+    networkConfig: NetworkConfig,
+    signer: ethers.Signer,
+    address: string,
+    contributions: {
+      id: number;
+      activityTypeId: number;
+      userId: number;
+      args: MintArgs;
+      name: Uint8Array;
+      details: Uint8Array;
+      proof: Uint8Array;
+    }[],
+  ) {
+    const contract = new GovrnContract(networkConfig, signer);
+    const args = {
+      contributions: contributions.map(c => ({
+        contribution: {
+          owner: address,
+          detailsUri: c.args.detailsUri,
+          dateOfSubmission: c.args.dateOfSubmission,
+          dateOfEngagement: c.args.dateOfEngagement,
+        },
+      })),
+    };
+    const transaction = await contract.bulkMint(args);
+    const transactionReceipt = await transaction.wait(REQUIRED_CONFIRMATIONS);
+
+    const onChainIds: { [index: number]: ethers.BigNumber | null } =
+      transactionReceipt.logs
+        .map(l => {
+          console.log('on chain id (logs)', l);
+          return contract.govrn.interface.parseLog(l);
+        })
+        .reduce(
+          (prev, current, idx) => ({
+            ...prev,
+            [idx]: current.name === 'Mint' ? current.args['id'] : null,
+          }),
+          {},
+        );
+
+    return await Promise.allSettled(
```

Comment:
My worry with this is that we trigger all the requests at once and if there are a lot like 100s then that would put a lot of strain on the backend. It is probably best we we can batch these calls and wait for a batch to finish.